### PR TITLE
feat: Change coin gecko api for price api on TokenRates Controller

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
This PR aims to change the coin gecko api for our new price api.

On this PR we have:
* Changed mapping for a `for of` to not get the chance of ending on a rate limit error (we do requests in chunks of 100 tokens)
* The pooling update takes 30 minutes, on an expectation to lower the network traffic
* Removed everything related to coin gecko from the token rates controller
* Implemented hardcoded list of currencies and chains supported of priceAPI
* Changed shiden network to the right chain id `336`

Core branch of the tokenRatesController patch: `price-api-token-rates-controller` -> https://github.com/MetaMask/core/tree/mobile/price-api-token-rates-controller
## **Related issues**

Fixes:

## **Manual testing steps**

The test manual steps should include on multiple networks:
* Import detected tokens https://recordit.co/VkfM8nJulA
* Import auto-complete token https://recordit.co/6SXWsGSSFn
* Import manual token https://recordit.co/6SXWsGSSFn

It should show the value of the tokens.

## **Screenshots/Recordings**
https://recordit.co/VkfM8nJulA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: patches/@metamask+assets-controllers+7.0.0.patch [hunk 1]
```diff
@@ -883,19 +1317,23 @@ index e3f81e9..10994ef 100644
...
++                        this.fetchExchangeRate(controller_utils_1.FALL_BACK_VS_CURRENCY),
                          (0, crypto_compare_1.fetchExchangeRate)(nativeCurrency, controller_utils_1.FALL_BACK_VS_CURRENCY, false),
                      ]);
                  }
-@@ -270,6 +325,7 @@ class TokenRatesController extends base_controller_1.BaseController {
+@@ -270,8 +599,10 @@ class TokenRatesController extends base_controller_1.BaseController {
                      }
                      throw error;
                  }
 +                const tokenExchangeRates = Object.assign({}, ...tokenExchangeRatesResponse);
```
Comment (Gudahtt) [inline patches/@metamask+assets-controllers+7.0.0.patch:1328]: Hmm, what's going on here? Not sure on the purpose of this shallow clone + typecast.

Comment (tommasini) [reply patches/@metamask+assets-controllers+7.0.0.patch:1328]: I'm changing now to the latest version right now, but the purpose of this line it was that tokenExchangeRatesResponse it is an array, and too keep the old logic I just convert it back to an object

**Comment by tommasini** [comment]
Converted this PR to a draft since we built a new one with the latest TokenRatesController version that already have into account the price api 
https://github.com/MetaMask/metamask-mobile/pull/8143/files

**Comment by tommasini** [comment]
closing this since this is already shipped

**Pull request closed without merging**